### PR TITLE
Implement `RRule::Rule#humanize`

### PR DESCRIPTION
**Pull request by craigmcnamara**
I got jealous of https://github.com/jakubroztocil/rrule being able to print a human understandable version of RRule and decided to take a run at making it work.

Are you at all interested in this or should I keep it in my own fork? This is a bit incomplete, but it's actually all I need for my app. If you'd take the pull request I'd go through the motions to complete the port.

### CODE REVIEW

File: lib/rrule/weekday.rb [hunk 1]
```diff
@@ -4,6 +4,16 @@ module RRule
   class Weekday
     attr_reader :index, :ordinal
 
+    DAY_NAMES = %w[
```
Comment (rmitchell-sq) [inline lib/rrule/weekday.rb:7]: Possible to move these new methods to the `Humanizer` class? I think it'd be nice to have all the code that deals with humanizing rules in one place.

**Comment by rmitchell-sq** [comment]
> I got jealous of https://github.com/jakubroztocil/rrule being able to print a human understandable version of RRule and decided to take a run at making it work.
> 
> Are you at all interested in this or should I keep it in my own fork? This is a bit incomplete, but it's actually all I need for my app. If you'd take the pull request I'd go through the motions to complete the port.

Sure, happy to add this, just left one comment.

Ideally this would be i18n-able, but I can imagine that'd be more complicated than just swapping out the words, and that non-English languages likely have different sentence structures that change the way you need to humanize the rule, so fine to leave that for a potential future improvement.

**Comment by craigmcnamara** [comment]
Ok I think it's ready for a review and merge. The highlights are

- Humanizer is self contained now.
- No more Seattle Style or JS idioms to make RuboCop happy.
- Implemented more cases that can be humanized.

File: lib/rrule/weekday.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ def initialize(index, ordinal = nil)
     def self.parse(weekday)
       match = /([+-]?\d+)?([A-Z]{2})/.match(weekday)
       index = RRule::WEEKDAYS.index(match[2])
-      ordinal = match[1] ? match[1].to_i : nil
+      ordinal = match[1]&.to_i
```
Comment (craigmcnamara) [inline lib/rrule/weekday.rb:15]: I didn't mean to change this, but RuboCop was complaining about it.

**Comment by craigmcnamara** [comment]
> Ideally this would be i18n-able, but I can imagine that'd be more complicated than just swapping out the words, and that non-English languages likely have different sentence structures that change the way you need to humanize the rule, so fine to leave that for a potential future improvement.

I now have a morbid curiosity about what it would take to make this string slamming be more in line with i18n. I'll be looking in to that a bit more.

**Comment by rmitchell-sq** [comment]
LGTM, thanks for the PR!

**Comment by bahriddin** [comment]
Thanks for the great gem and nice PR 👏🏼
Are you planning to release a new tag with latest changes?
Regards!

**Comment by rmitchell-sq** [comment]
> Thanks for the great gem and nice PR 👏🏼 Are you planning to release a new tag with latest changes? Regards!

Yep, makes sense - maybe once https://github.com/square/ruby-rrule/pull/55 lands, just to get all caught up?

**Comment by rmitchell-sq** [comment]
> Thanks for the great gem and nice PR 👏🏼 Are you planning to release a new tag with latest changes? Regards!

Released as [0.6.0](https://rubygems.org/gems/rrule/versions/0.6.0) 🎉

**Pull request merged**